File: go/unit_test/demo/calculator_test.go
package demo

import "testing"

func TestAdd(t *testing.T) {
	result := Add(2, 3)
	expected := 5
	if result != expected {
		t.Errorf("Add(2, 3) = %d; want %d", result, expected)
	}
}

func TestSubtract(t *testing.T) {
	result := Subtract(5, 3)
	expected := 2
	if result != expected {
		t.Errorf("Subtract(5, 3) = %d; want %d", result, expected)
	}
}

File: ruby/unit_test/demo/calculator_test.rb
require 'minitest/autorun'
require_relative 'calculator'

class TestCalculator < Minitest::Test
  def setup
    @calculator = Calculator.new
  end

  def test_add
    puts "\nTesting addition method..."
    assert_equal 5, @calculator.add(2, 3)
  end

  def test_subtract
    puts "\nTesting subtraction method..."
    assert_equal 2, @calculator.subtract(5, 3)
  end
end